scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -2656,10 +2656,11 @@ public Builder setCertificateVerifier(CertificateVerifier verifier) {
 				throw new NullPointerException("CertificateVerifier must not be null");
 			} else if (config.trustStore != null) {
 				throw new IllegalStateException("CertificateVerifier must not be used after trust store is set!");
+			} else if (config.advancedCertificateVerifier != null) {
+				throw new IllegalStateException(
+						"CertificateVerifier must not be used after new certificate verifier is set!");
 			}
 			config.certificateVerifier = verifier;
-			config.advancedCertificateVerifier = BridgeCertificateVerifier.builder()
-					.setCertificateVerifier(verifier).build();
```

Comment:
You are talking about removing this line ?

> also fails the checks in the unit-tests like in ConnectionHelper:

I'm not sure to get the issue. I run mvn clean install and I don't see any unit-tests failing ?
Did I missed something ?

Reply:
Yes.
And yes, the unit test are not failing. Otherwise I would not have given LGTM.
And for now, no test is using the trust store nor the rpk trust store.

If some test are accidentally modified or added in the future using the setter for them (even if they are deprecated), then the result may be surprising.

Reply:
I can add support of  `config.certificateVerifier != null` if you want.
But this means allowing to use `setCertificateVerifier` with `null` which is not allowed currently  like it was not allowed for RPKStore and trustStore. (I see you add it recently when you add new advanced certificate verifier).

Reply:
No, it was just a remark, if unexpected, larger bugfixes gets required.